src/plugins/generic/sqlitehistory/sqlitehistory.cpp

```diff
@@ -0,0 +1,506 @@
+/****************************************************************************
+**
+** qutIM - instant messenger
+**
+** Copyright © 2015 Nicolay Izoderov <nico-izo@ya.ru>
+**
+*****************************************************************************
+**
+** $QUTIM_BEGIN_LICENSE$
+** This program is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This program is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+** See the GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see http://www.gnu.org/licenses/.
+** $QUTIM_END_LICENSE$
+**
+****************************************************************************/
+
+#include "sqlitehistory.h"
+#include <qutim/chatunit.h>
+#include <qutim/account.h>
+#include <qutim/protocol.h>
+#include <qutim/systeminfo.h>
+#include <QStringBuilder>
+#include <QThreadPool>
+#include <qutim/icon.h>
+#include <qutim/debug.h>
+//#include <QElapsedTimer>
+#include <QQueue>
+#include <QtSql/QSqlError>
+#include <QRegularExpression>
+
+namespace Core
+{
+
+SqliteHistory::SqliteHistory() :
+	m_thread(new QThread),
+	m_worker(new SqliteWorker)
+{
+
+	m_worker->moveToThread(m_thread);
+
+	connect(m_worker, SIGNAL(error(QString)), this, SLOT(errorHandler(QString)));
+
+	connect(m_thread, SIGNAL(started()), m_worker, SLOT(process()));
+
+	connect(m_worker, SIGNAL(finished()), m_thread, SLOT(quit()));
+	connect(m_worker, SIGNAL(finished()), m_worker, SLOT(deleteLater()));
+	connect(m_thread, SIGNAL(finished()), m_thread, SLOT(deleteLater()));
+
+	m_thread->start();
```

Comment:
Each initialization must have a deinitialization, may you add a m_thread->join() to destructor?